include/alpaka/core/ApiCudaRt.hpp

```diff
@@ -0,0 +1,357 @@
+            cudaStream_t stream)
+        {
+            return ::cudaMemcpy2DAsync(dst, dpitch, src, spitch, width, height, kind, stream);
+        }
+
+        static inline cudaError_t memcpy3DAsync(const struct cudaMemcpy3DParms* p, cudaStream_t stream)
+        {
+            return ::cudaMemcpy3DAsync(p, stream);
+        }
+
+        static inline cudaError_t memcpyAsync(
+            void* dst,
+            const void* src,
+            size_t count,
+            enum cudaMemcpyKind kind,
+            cudaStream_t stream)
+        {
+            return ::cudaMemcpyAsync(dst, src, count, kind, stream);
+        }
+
+        static inline cudaError_t memset2DAsync(
+            void* devPtr,
+            size_t pitch,
+            int value,
+            size_t width,
+            size_t height,
+            cudaStream_t stream)
+        {
+            return ::cudaMemset2DAsync(devPtr, pitch, value, width, height, stream);
+        }
+
+        static inline cudaError_t memset3DAsync(
+            struct cudaPitchedPtr pitchedDevPtr,
+            int value,
+            struct cudaExtent extent,
+            cudaStream_t stream)
+        {
+            return ::cudaMemset3DAsync(pitchedDevPtr, value, extent, stream);
+        }
+
+        static inline cudaError_t memsetAsync(void* devPtr, int value, size_t count, cudaStream_t stream)
+        {
+            return ::cudaMemsetAsync(devPtr, value, count, stream);
+        }
+
+        static inline cudaError_t setDevice(int device)
+        {
+            return ::cudaSetDevice(device);
+        }
+
+        static inline cudaError_t streamAddCallback(
+            cudaStream_t stream,
+            cudaStreamCallback_t callback,
+            void* userData,
+            unsigned int flags)
+        {
+            return ::cudaStreamAddCallback(stream, callback, userData, flags);
+        }
+
+        static inline cudaError_t streamCreate(cudaStream_t* pStream)
+        {
+            return ::cudaStreamCreate(pStream);
+        }
+
+        static inline cudaError_t streamCreateWithFlags(cudaStream_t* pStream, unsigned int flags)
+        {
+            return ::cudaStreamCreateWithFlags(pStream, flags);
+        }
+
+        static inline cudaError_t streamDestroy(cudaStream_t stream)
+        {
+            return ::cudaStreamDestroy(stream);
+        }
+
+        static inline cudaError_t streamQuery(cudaStream_t stream)
+        {
+            return ::cudaStreamQuery(stream);
+        }
+
+        static inline cudaError_t streamSynchronize(cudaStream_t stream)
+        {
+            return ::cudaStreamSynchronize(stream);
+        }
+
+        static inline cudaError_t streamWaitEvent(cudaStream_t stream, cudaEvent_t event, unsigned int flags)
+        {
+            return ::cudaStreamWaitEvent(stream, event, flags);
+        }
+
+        // device API
+        static inline struct cudaPitchedPtr makePitchedPtr(void* d, size_t p, size_t xsz, size_t ysz)
```

Comment:
If this is a device function, don't we need `__device__`?

Reply:
No, they are actually host functions - I don't remember where I copied the (wrong) comment from.